internal/services/network/application_gateway_resource.go

```diff
@@ -3223,10 +3239,16 @@ func flattenApplicationGatewayRewriteRuleSets(input *[]network.ApplicationGatewa
 
 							if config.ModifiedPath != nil {
 								urlConfig["path"] = *config.ModifiedPath
+								urlConfig["components"] = "path_only"
 							}
 
 							if config.ModifiedQueryString != nil {
 								urlConfig["query_string"] = *config.ModifiedQueryString
+								urlConfig["components"] = "query_string_only"
+							}
+
+							if config.ModifiedQueryString != nil && config.ModifiedPath != nil {
+								urlConfig["components"] = ""
 							}
```

Comment:
presumably `components` should be defaulted to `""` and then overridden if either Path or QueryString isn't present?